piksi_tools/console/imu_view.py

```diff
@@ -79,25 +95,30 @@ def imu_aux_callback(self, sbp_msg, **metadata):
             print("IMU type %d not known" % sbp_msg.imu_type)
 
     def imu_raw_callback(self, sbp_msg, **metadata):
-        self.acc[:-1, :] = self.acc[1:, :]
-        self.gyro[:-1, :] = self.gyro[1:, :]
-        self.acc[-1] = (sbp_msg.acc_x, sbp_msg.acc_y, sbp_msg.acc_z)
-        self.gyro[-1] = (sbp_msg.gyr_x, sbp_msg.gyr_y, sbp_msg.gyr_z)
-
-        if time.time() - self.last_plot_update_time > GUI_UPDATE_PERIOD:
-            self.imu_set_data()
-            if self.imu_conf is not None:
-                acc_range = self.imu_conf & 0xF
-                sf = 2. ** (acc_range + 1) / 2. ** 15
-                self.rms_acc_x = sf * np.sqrt(np.mean(np.square(self.acc[:, 0])))
-                self.rms_acc_y = sf * np.sqrt(np.mean(np.square(self.acc[:, 1])))
-                self.rms_acc_z = sf * np.sqrt(np.mean(np.square(self.acc[:, 2])))
+        memoryview(self.acc_x)[:-1] = memoryview(self.acc_x)[1:]
```

Comment:
just a question: did you notice performance improvements from using the memoryview?

Reply:
Not that I could measure (or perceive), but in theory it should be faster since there's no copying involved.